### ISSUE TITLE
Release problem with 0.9.24?

### ISSUE THREAD

**Issue by vincenzobaz** [creativescala/doodle#102]
Version `0.9.24` seems to be incompletely published:
```
➜  ~ cs complete org.creativescala:doodle_3:0.9.24  
➜  ~ cs complete org.creativescala:doodle_3:0.9.2 
➜  ~ cs complete org.creativescala:doodle_3:     
➜  ~ cs complete org.creativescala:doodle_2.13
doodle_2.13
➜  ~ cs complete org.creativescala:doodle_2.13:
0.9.16
0.9.17
0.9.18
0.9.19
0.9.20
0.9.21
0.9.22
0.9.23
```
If I use [Maven's Central web UI](https://search.maven.org/search?q=g:org.creativescala) then for this version I only see Scala JS artifacts

**Comment by noelwelsh** [comment]
Thanks for letting me know. I'll take a look.

**Comment by noelwelsh** [comment]
I've changed the build and sbt seems happier, but I've run into an issue I have no idea how to fix (conflicting cross-version suffixes but I only have dependencies that are available for Scala 3) so :shrug:

**Comment by SethTisue** [comment]
You probably have a dependency that is available for Scala 3, but itself declared a dependency on a 2.13 artifact. What's the conflict you're seeing?

(`dependencyTree` might help, but might not: https://github.com/sbt/sbt/issues/6618)

**Comment by noelwelsh** [comment]
The error seems to occur when publishing the 2.13 build.

E.g. If I run  `++ 2.13.6 publishLocal` amongst all the output is the following error

```
[error] Modules were resolved with conflicting cross-version suffixes in ProjectRef(uri("file:/home/noel/dev/creativescala/doodle/"), "rootJVM"):
[error]    org.typelevel:cats-free _3, _2.13
[error]    org.typelevel:cats-effect _3, _2.13
[error]    org.typelevel:simulacrum-scalafix-annotations _3, _2.13
[error]    org.typelevel:cats-kernel _3, _2.13
[error]    org.typelevel:cats-core _3, _2.13
```

I seem the same error for the `rootJS` project. The sub-projects, which the root project aggregates, seem to be fine. The root project, following the sbt documentation, sets `crossScalaVersions` to `Nil`.

**Comment by SethTisue** [comment]
Weird. Okay, this is something different than I expected. You need an expert on sbt-scalajs-crossproject, which I'm not. I have not run into this.

**Comment by SethTisue** [comment]
Ah, this is something different than I expected.

After `++2.13.6`, `rootJVM/update` is sufficient to reproduce the problem.

I think you need to either:

* not include `commonSettings` in `root`
* don't set `libraryDependencies` in `commonSettings`

Maybe you could add the dependencies on cats-effect et al to `coreJVM` only, since the other projects depend on it? That's how I would normally write it in the first place, actually.

**Comment by noelwelsh** [comment]
Thanks for that suggestion. It seems to work. I've published 0.9.25.
